src/handleInteraction.ts

```diff
@@ -0,0 +1,192 @@
+import type { CommandInteraction, DMChannel, GuildMember, GuildTextBasedChannel } from 'discord.js';
+import { allCommands } from './commands';
+import { ChannelType } from 'discord.js';
+import { isGuildedCommandContext } from './helpers/guards/isGuildedCommandContext';
+import { logUser } from './helpers/logUser';
+import { replyPrivately, sendMessageInChannel } from './helpers/actions/messages/replyToMessage';
+
+/**
+ * Performs actions from a Discord command interaction.
+ * The command is ignored if the interaction is from a bot.
+ *
+ * Constructs a context object to send to command executors.
+ * This context handles many edge cases and gotchas with interactions,
+ * including follow-up responses, long-running commands,
+ * typing indicators, ephemeral replies, partial fields,
+ * and error handling.
+ *
+ * The goal is for us devs to not have to worry about how exactly
+ * things get done when we're writing command handlers, only
+ * that what we say goes.
+ *
+ * @param logger The place to write system messages.
+ */
+export async function handleInteraction(
+	interaction: CommandInteraction,
+	logger: Console
+): Promise<void> {
+	// Don't respond to bots or ourselves
+	if (interaction.user.bot) return;
+	if (interaction.user.id === interaction.client.user.id) return;
+
```

Comment:
I'd recommend adding
```suggestion
if (!interaction.isChatInputCommand()) return;
```
because not every interaction is a command.

Reply:
Nevermind, you did this elsewhere.